src/main/java/org/folio/rest/helper/TransactionsHelper.java

```diff
@@ -67,4 +74,36 @@ private Transaction modifyTransaction(Transaction transaction, AwaitingPayment a
     return transaction;
   }
 
+  public CompletableFuture<Transaction> createEncumbrance(Transaction encumbrance) {
+    return checkEncumbranceRestrictions(encumbrance)
+      .thenCompose(v -> createTransaction(encumbrance));
+  }
+
+  public CompletableFuture<Void> checkEncumbranceRestrictions(Transaction encumbrance) {
+    String budgetQuery = "fundId==" + encumbrance.getFromFundId() + " AND fiscalYearId==" + encumbrance.getFiscalYearId();
+    return new BudgetsHelper(httpClient, okapiHeaders, ctx, lang).getSingleBudgetByQuery(budgetQuery)
+      .thenCompose(budget -> {
+        if (budget.getBudgetStatus() == Budget.BudgetStatus.ACTIVE) {
+          String ledgerQuery = "fund.id==" + encumbrance.getFromFundId() + " AND fiscalYear.id==" + encumbrance.getFiscalYearId();
+          return new LedgersHelper(httpClient, okapiHeaders, ctx, lang).getSingleLedgerByQuery(ledgerQuery)
+            .thenAccept(ledger -> checkLedgerEncumbranceRestrictions(encumbrance, budget, ledger));
+        } else {
+          throw new HttpException(422, BUDGET_IS_INACTIVE);
+        }
+      });
+  }
+
+  private void checkLedgerEncumbranceRestrictions(Transaction encumbrance, Budget budget, Ledger ledger) {
+    if (ledger.getRestrictEncumbrance() && budget.getAllowableEncumbrance() != null) {
+      Double availableForEncumbrance = getBudgetAvailableForEncumbrance(budget.getAvailable(), budget.getAllowableEncumbrance(), encumbrance.getCurrency());
+      if (encumbrance.getAmount() > availableForEncumbrance) {
```

Comment:
It is not clear from the WIKI how to calculate Remaining Allowable Encumbrance. @craigmcnally couldn't you shed some light on that?

Reply:
it's on the wiki...

> The "Remaining Allowable Enc." in the flowchar is calculated as follows:
>
>[remaining amount we can encumber] = (allocated * allowableEncumbered) - (allocated - (unavailable + available)) - (encumbered + awaitingPayment)
>
>The "restrictEncumbrances" flag actually lives in the ledger, so we'll need to find traverse:  budget → fund → ledger.restrictEncumbrances